chroma_core/services/job_scheduler/job_scheduler.py

```diff
@@ -1745,3 +1748,106 @@ def update_lnet_configuration(self, lnet_configuration_list):
     @property
     def CommandPlan(self):
         return CommandPlan(self._lock_cache, self._job_collection)
+
+    def configure_stratagem(self, stratagem_data):
+        with self._lock:
```

Comment:
What is the lock doing here?

Reply:
I think in every other place where we are adding to the cache it is being done inside a lock. Should I remove the lock?